soarca_fin_python_library/executor.py

```diff
@@ -105,15 +105,15 @@ def _handle_unregister_self_message(self, message: UnregisterSelf):
                 self._wait_for_ack(message.message_id)
                 self.callback(message)
                 return
-            except (Empty, TimeoutError) as e:
+            except (Empty, TimeoutError):
                 self._send_message_as_json(message, topic="soarca")
-            except RuntimeError as e:
+            except RuntimeError:
                 self._send_message_as_json(message, topic="soarca")
             finally:
                 retries -= 1
                 if retries == 0:
                     log.error(
-                        f"Did not receive an ack for message {message.message_id}. Aborting...")
+                        "Did not receive an ack for message %s. Aborting...", message.message_id)
                     exit(-1)
```

Comment:
Exit after communication failure is a bit hard on the user maybe error mor gracefully and keep running. You can discard any context at this point and prepare for a new command.

Reply:
The functions where the exists are called are in register and unregister. I would argue that when a register fails after 3 tries the program should exit. The other exit is called when the fin itself is trying to unregister, but it did not receive a ack back from SOARCA. I would argue that the program should exit aswell.